### PR TITLE
Struct serialization

### PR DESCRIPTION
**Pull request by netoneko**
## Description

- [x] serialize a struct
- [x] deserialize a struct
- [x] delete a struct
- [x] rename a struct

## Types of changes
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have read the [CONTRIBUTING](https://github.com/orbs-network/orbs-contributor-guide/blob/master/CONTRIBUTING.md) doc
- [ ] I have opened an issue and discussed the solution suggested in this PR
- [ ] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

@jlevison @noambergIL @ronnno please let me know if you think the API should look different

### CODE REVIEW

File: go/sdk/v1/state/rename.go [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package state
+
+func Rename(oldKey []byte, newKey []byte) {
```
Comment (jlevison) [inline go/sdk/v1/state/rename.go]: added a test that checks renaming from a non-existing key does not blow up

Comment (netoneko) [reply go/sdk/v1/state/rename.go]: Done.

**Comment by netoneko** [comment]
The test fails because the e2e relies on this SDK being merged into orbs-network-go, which is a circular dependency.

**Comment by netoneko** [comment]
Declined in favor of creating a new repository: https://github.com/orbs-network/contract-external-libraries-go

**Pull request closed without merging**